lib/spack/spack/test/concretize.py

```diff
@@ -350,29 +350,41 @@ def test_compiler_flags_differ_identical_compilers(self):
             assert spec.satisfies("cflags=-O2")
 
     def test_concretize_compiler_flag_propagate(self):
-        spec = Spec("hypre cflags=='-g' ^openblas")
+        spec = Spec("callpath cflags=='-g'")
         spec.concretize()
 
-        assert spec.satisfies("^openblas cflags='-g'")
+        assert spec.satisfies("^dyninst cflags='-g'")
+        assert spec.satisfies("^libelf cflags='-g'")
 
     @pytest.mark.only_clingo(
         "Optional compiler propagation isn't deprecated for original concretizer"
     )
     def test_concretize_compiler_flag_does_not_propagate(self):
-        spec = Spec("hypre cflags='-g' ^openblas")
+        spec = Spec("callpath cflags='-g'")
         spec.concretize()
 
-        assert not spec.satisfies("^openblas cflags='-g'")
+        assert not spec.satisfies("^dyninst cflags='-g'")
 
     @pytest.mark.only_clingo(
         "Optional compiler propagation isn't deprecated for original concretizer"
     )
     def test_concretize_propagate_compiler_flag_not_passed_to_dependent(self):
-        spec = Spec("hypre cflags=='-g' ^openblas cflags='-O3'")
+        spec = Spec("callpath cflags=='-g' ^dyninst cflags='-O3'")
         spec.concretize()
 
         assert set(spec.compiler_flags["cflags"]) == set(["-g"])
-        assert spec.satisfies("^openblas cflags='-O3'")
+        assert spec.satisfies("^dyninst cflags='-O3'")
+        assert spec.satisfies("^libelf cflags='-g'")
+
+    @pytest.mark.only_clingo(
+        "Optional compiler propagation isn't deprecated for original concretizer"
+    )
+    def test_concretize_propagate_specified_compiler_flag(self):
+        spec = Spec("callpath cflags=='-g' cxxflags='-O3'")
+        spec.concretize()
+
+        assert spec.satisfies("^dyninst cflags='-g'")
+        assert not spec.satisfies("^dyninst cxxflags='-O3'")
```

Comment:
```suggestion
        for dep in spec.traverse(root=False):
            assert spec.satisfies("cflags='-g'")
            assert not spec.satisfies("cxxflags='-O3'")
```
Better if we test every node and also if we don't make the test care whether the dependencies change.